### PR TITLE
Add dispatch on parametric types in doc

### PR DESCRIPTION
**Pull request by matthieugomez**
[av-skip] This issue with type invariance often comes up.

### CODE REVIEW

File: doc/manual/types.rst [hunk 1]
```diff
@@ -800,6 +800,20 @@ Constructor methods to appropriately handle such mixed cases can be
 defined, but that will not be discussed until later on in
 :ref:`man-constructors`.
 
+Due to type invariance, the following method does not accept arguments of type ``Point{Float64}``::
+
+    function norm(p::Point{Real})
+       sqrt(p.x^2 + p.y^2)
+    end
+
+The correct way to define a method that accepts all arguments of ``Point`` types where ``T`` is a subtype of ``Real`` is::
```
Comment (nalimilan) [inline doc/manual/types.rst]: "of type ``Point``", maybe?

**Comment by johnmyleswhite** [comment]
The text is really great, but there probably needs to be more of a transition into this new topic, which seems distinct from the constructors discussed immediately above.

File: doc/manual/types.rst [hunk 2]
```diff
@@ -800,6 +800,20 @@ Constructor methods to appropriately handle such mixed cases can be
...
+
+    function norm{T <: Real}(p::Point{T})
```
Comment (nalimilan) [inline doc/manual/types.rst]: It looks like the convention is to write `{T<:Real}` to me (though it's hard to be sure...).

**Comment by nalimilan** [comment]
Good idea!

**Comment by matthieugomez** [comment]
Thanks for the comments — updated the pull request.

**Comment by tkelman** [comment]
backported in 7048c4007915afd935816423ecece7b4c37ed8dc

**Pull request merged**